library/alloc/src/sync.rs

```diff
@@ -328,6 +328,79 @@ impl<T> Arc<T> {
         Self::from_inner(Box::leak(x).into())
     }
 
+    /// Constructs a new `Arc<T>` using a weak reference to itself. Attempting
+  /// to upgrade the weak reference before this function returns will result
```

Comment:
I think the indentation for docs is off starting here.

Reply:
That's odd, is this not just cherry-picking the commits of https://github.com/rust-lang/rust/pull/72443?

Reply:
cherry picking took a while due to some conflicts since this pr was small, I pasted the contents directly into respectively files. I ran tidy again now which would have fixed this issues